tests/ui/async-await/in-trait/async-generics-and-bounds.rs

```diff
@@ -1,12 +1,13 @@
 //@ check-fail
-//@ known-bug: #102682
 //@ edition: 2021
```

Comment:
Also, idk why this issue was closed. This definitely still represents an issue with async fn in general, where we don't deduce the outlives from the return type:

```
async fn foo<'a, T>() -> &'a T { todo!() }
```

(because we don't walk into RPITs, i.e. we don't deduce any implied outlives bounds from `impl Future<Output = &'a T>`).